src/main/java/com/desiato/puresynth/configurations/AuthTokenFilter.java

```diff
@@ -0,0 +1,63 @@
+package com.desiato.puresynth.configurations;
+
+import com.desiato.puresynth.dtos.PureSynthToken;
+import com.desiato.puresynth.services.AuthenticationService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+public class AuthTokenFilter extends OncePerRequestFilter {
+
+    private final AuthenticationService authenticationService;
+
+    public AuthTokenFilter(AuthenticationService authenticationService){
+        this.authenticationService = authenticationService;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String tokenValue = request.getHeader("authToken");
+
+        if (isPublicEndpoint(request.getRequestURI())) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        if (tokenValue == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            response.getWriter().write("Unauthorized");
+            return;
+        }
+
+        try {
+            PureSynthToken pureSynthToken = new PureSynthToken(tokenValue);
+            authenticationService.createUserDetails(pureSynthToken)
+                    .map(userDetails -> new UsernamePasswordAuthenticationToken(userDetails, null, userDetails.getAuthorities()))
+                    .ifPresent(authentication -> {
+                        SecurityContextHolder.getContext().setAuthentication(authentication);
+                        try {
+                            filterChain.doFilter(request, response);
+                        } catch (IOException | ServletException e) {
```

Comment:
Why are you catching this and rapping in the RTE? It hides a potential problem.

Reply:
Yes, I was not thinking about the unchecked exception issue. I modified the implementation, please let me know if this works

Reply:
No, this is also bad approach. At this place we can't handle those exception, we should let them throw. It seems, that in Java there[ is no good way](https://stackoverflow.com/questions/18198176/java-8-lambda-function-that-throws-exception) to handle checked exception in lambda.

The easiest would be to go back to:
```
var authenticationDetails = authenticationService.createUserDetails(pureSynthToken)
                    .map(userDetails -> new UsernamePasswordAuthenticationToken(userDetails, null, userDetails.getAuthorities()));
            if (authenticationDetails.isPresent()) {
                SecurityContextHolder.getContext().setAuthentication(authenticationDetails.get());
                filterChain.doFilter(request, response);
            }
```